base_layer/wallet/src/storage/sqlite_db/models.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2025 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+use chacha20poly1305::XChaCha20Poly1305;
+use tari_common_types::{
+    burn_proof::{BurnClaimProof, EncodedMerkleProof},
+    encryption::{decrypt_bytes_integral_nonce, encrypt_bytes_integral_nonce, Encryptable},
+    types::FixedHash,
+};
+use tari_transaction_components::transaction_components::TransactionKernel;
+use tari_utilities::Hidden;
+
+use crate::{error::WalletStorageError, schema, storage::serializers};
+
+pub struct DbBurnProof {
+    pub id: i32,
+    pub output_hash: FixedHash,
+    pub burn_proof: BurnClaimProof,
+    pub kernel: TransactionKernel,
+    pub kernel_merkle_proof: Option<EncodedMerkleProof>,
+    pub created_at: chrono::NaiveDateTime,
+    pub updated_at: chrono::NaiveDateTime,
+}
+
+impl TryFrom<BurntProofSql> for DbBurnProof {
+    type Error = WalletStorageError;
+
+    fn try_from(value: BurntProofSql) -> Result<Self, Self::Error> {
+        let burn_proof = serializers::bincode_decode(&value.burn_proof)?;
+        let kernel = serializers::bincode_decode(&value.kernel)?;
+        let kernel_merkle_proof = value
+            .kernel_merkle_proof
+            .as_ref()
+            .map(|kp| serializers::bincode_decode(kp))
+            .transpose()?;
+        let output_hash = FixedHash::try_from(value.output_hash.as_slice())
+            .map_err(|e| WalletStorageError::ConversionError(format!("Invalid output hash length in DB: {}", e)))?;
+        Ok(Self {
+            id: value.id,
+            output_hash,
+            burn_proof,
+            kernel,
+            kernel_merkle_proof,
+            created_at: value.created_at,
+            updated_at: value.updated_at,
+        })
+    }
+}
+
+#[derive(Debug, Queryable)]
+#[diesel(table_name = schema::burn_proofs)]
+pub(crate) struct BurntProofSql {
+    pub id: i32,
+    pub output_hash: Vec<u8>,
+    pub commitment: Vec<u8>,
+    pub burn_proof: Vec<u8>,
+    pub kernel: Vec<u8>,
+    pub kernel_merkle_proof: Option<Vec<u8>>,
+    pub created_at: chrono::NaiveDateTime,
+    pub updated_at: chrono::NaiveDateTime,
+}
+
+fn encrypt_domain(commitment: &[u8], field_name: &'static str) -> Vec<u8> {
+    [BurntProofSql::BURNT_PROOF, commitment, field_name.as_bytes()].concat()
+}
+
+impl Encryptable<XChaCha20Poly1305> for BurntProofSql {
+    fn domain(&self, field_name: &'static str) -> Vec<u8> {
+        encrypt_domain(&self.commitment, field_name)
+    }
+
+    fn encrypt(mut self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
+        self.burn_proof =
+            encrypt_bytes_integral_nonce(cipher, self.domain("encoded_burn_proof"), Hidden::hide(self.burn_proof))?;
```

Comment:
It feels like `"encoded_burn_proof"` should be a constant.
```suggestion
            encrypt_bytes_integral_nonce(cipher, self.domain("encoded_burn_proof"), Hidden::hide(self.burn_proof))?;
```

Reply:
yeah, I'm just copying what is in the rest of the wallet - remember this wallet is bring rewritten